splunk/splunk_test.go

```diff
@@ -0,0 +1,64 @@
+package splunk
+
+import (
+	_ "embed"
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/redhat-appstudio/segment-bridge.git/containerfixture"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+//go:embed splunk_container_template.yaml
+var manifest string
+
+func TestGetRecordsCount(t *testing.T) {
+	_, callerFilePath, _, _ := runtime.Caller(0)
+	callerDirPath := filepath.Dir(callerFilePath)
+	netrcPath := fmt.Sprintf("%s/.netrc", callerDirPath)
+	fmt.Println(netrcPath)
+	
+
```

Comment:
Achievable, but unless we run everything through this fixture function (is this what we want?) , we'll need to defer the cleanUp function outside of it, since we need the service open until after this function is done running.

Reply:
disregard my last comment

Reply:
I'm not completely sure how to assemble this in a way that'll run the fixture only once during a parametrize test (is there a reason to deploy multiple pods just for another testcase in the same test?)

Reply:
Think of this like a Python context manager - in Python when you create a resource that you need to clean up, you make a context manager to manage that and you wrap everything that needs the resource in it.

In Go you achieve pretty much the same via a function that defers the cleanup an calls another closure/function that is passed to it.

WRT the test parameter loops - you can either wrap the entire loop with the "context manager" or wrap each individual subtest with it, depending on wither you want to clean the resources between subtests.